packages/wkt/test/wkt.test.js

```diff
@@ -604,3 +604,404 @@ test('should parse a MULTIPOLYGON with a hole', function (t) {
+        coordinates: [40, 10]
+      },
+      {
+        type: 'LineString',
+        coordinates: [
+          [10, 10],
+          [20, 20],
+          [10, 40]
+        ]
+      },
+      {
+        type: 'Polygon',
+        coordinates: [
+          [
+            [40, 40],
+            [20, 45],
+            [45, 30],
+            [40, 40]
+          ]
+        ]
+      }
+    ],
+    properties: {
+      z: true
+    }
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with M property', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION M ( POINT (40 10), LINESTRING (10 10, 20 20, 10 40), POLYGON ((40 40, 20 45, 45 30, 40 40)))';
+
+  const output = wktToGeoJSON(input);
+
+  t.deepEqual(output, {
+    type: 'GeometryCollection',
+    geometries: [
+      {
+        type: 'Point',
+        coordinates: [40, 10]
+      },
+      {
+        type: 'LineString',
+        coordinates: [
+          [10, 10],
+          [20, 20],
+          [10, 40]
+        ]
+      },
+      {
+        type: 'Polygon',
+        coordinates: [
+          [
+            [40, 40],
+            [20, 45],
+            [45, 30],
+            [40, 40]
+          ]
+        ]
+      }
+    ],
+    properties: {
+      m: true
+    }
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with geometry collection embedded', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION ( POINT (40 10), LINESTRING (10 10, 20 20, 10 40), POLYGON ((40 40, 20 45, 45 30, 40 40)), GEOMETRYCOLLECTION ( POINT (40 10), LINESTRING (10 10, 20 20, 10 40), POLYGON ((40 40, 20 45, 45 30, 40 40))))';
+
+  const output = wktToGeoJSON(input);
+
+  t.deepEqual(output, {
+    type: 'GeometryCollection',
+    geometries: [
+      {
+        type: 'Point',
+        coordinates: [40, 10]
+      },
+      {
+        type: 'LineString',
+        coordinates: [
+          [10, 10],
+          [20, 20],
+          [10, 40]
+        ]
+      },
+      {
+        type: 'Polygon',
+        coordinates: [
+          [
+            [40, 40],
+            [20, 45],
+            [45, 30],
+            [40, 40]
+          ]
+        ]
+      },
+      {
+        type: 'GeometryCollection',
+        geometries: [
+          {
+            type: 'Point',
+            coordinates: [40, 10]
+          },
+          {
+            type: 'LineString',
+            coordinates: [
+              [10, 10],
+              [20, 20],
+              [10, 40]
+            ]
+          },
+          {
+            type: 'Polygon',
+            coordinates: [
+              [
+                [40, 40],
+                [20, 45],
+                [45, 30],
+                [40, 40]
+              ]
+            ]
+          }
+        ]
+      }
+    ]
+  });
+});
+
+test('should parse a GEOMETRYCOLLECTION with ZM property', function (t) {
+  t.plan(1);
+
+  const input = 'GEOMETRYCOLLECTION ZM ( POINT (40 10), LINESTRING (10 10, 20 20, 10 40), POLYGON ((40 40, 20 45, 45 30, 40 40)))';
```

Comment:
same here. neither Z nor M values are actually present in the geometries.

according to the [GeoJSON RFC](https://www.rfc-editor.org/rfc/rfc7946#section-3.1.1), Z values are supported but M values are not, so ideally this parser would preserve the first and omit the second.

Reply:
I could achieve this by simply removing the fourth insert on line 56 of wkt.yy. But that sounds like breaking changes outside the scope of the PR. I am simply using existing parsing to conveniently handle collection parsing.

Reply:
> I could achieve this by simply removing the fourth insert on line 56 of wkt.yy

i see. that's something i would welcome in a PR of its own.